test/tests/api/v1_1/nodes_tests.py

```diff
@@ -92,21 +94,28 @@ def __post_workflow(self, id, graph_name, data):
             sleep(1)
             timeout -= 1
         return timeout
-
-    def check_compute_count(self):
+    
+    def get_undiscovered_compute_count(self):
         Nodes().nodes_get()
         nodes = self.__get_data()
-        count = 0
+        discovered_count = 0
         for n in nodes:
             type = n.get('type')
             if type == 'compute':
-                count += 1
-        return count
+                id = n.get('id')
+                status = self.__get_workflow_status(id)
+                if status != 'pending' and status != 'running':
```

Comment:
This is probably not safe.   You're assuming the status will be correct which we have seen in the past its not always the case.  The status in AMQP is probably more reliable.  Secondly,  it assumes the status is for discovery graph,  other tests run workflows against nodes.  so by the time you get to 2.0 test groups,  you could be looking at workflow status for other workflows executed in prior tests.   It would probably be safer to just go through each node and check if the catalog is fulfilled.